tests/conftest.py

```diff
@@ -183,6 +183,7 @@ def gprofiler_docker_image(docker_client: DockerClient) -> Iterable[Image]:
     yield docker_client.images.get("gprofiler")
 
 
+@lru_cache(maxsize=1024)
```

Comment:
I implemented a cache this way, another option would be to use `fixture(scope="session") on `application_docker_image` but I prefer to have an explicit cache